decoder/vaapidecoder_h264.cpp

```diff
@@ -1637,6 +1647,19 @@ YamiStatus VaapiDecoderH264::ensureContext(const SharedPtr<SPS>& sps)
             return status;
         // return YAMI_DECODE_FORMAT_CHANGE to info upper layer va context changed
         return YAMI_DECODE_FORMAT_CHANGE;
+    }else if(isResolutionChanged(sps)){
+        m_configBuffer.width = sps->frame_cropping_flag ? sps->m_cropRectWidth
```

Comment:
@lizhong1008 ,  Maybe we can not run "m_configBuffer.surfaceWidth = sps->m_width;" before "isDecodeContextChanged(sps)", because in function isDecodeContextChanged(),  "m_configBuffer.surfaceWidth < sps->m_width" is a condition to judge whether context changed or not.